### PR TITLE
system_high_availability_settings

### PR DESCRIPTION
**Pull request by LuminatiHD**

### CODE REVIEW

File: plugins/modules/system_high_availability_settings.py [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+        "synchronize_peer_ip": {"type": "str", "required": False},
+        "synchronize_config_to_ip": {"type": "str", "required": False},
+        "remote_system_username": {"type": "str", "required": False},
+        "remote_system_password": {"type": "str", "required": False, "no_log": True},
+        "services_to_synchronize": {
+            "type": "list",
+            "required": False,
+            "elements": "str",
+        },
+    }
```
Comment (KiLLuuuhh) [inline plugins/modules/system_high_availability_settings.py:335]: why arent' settings like `Disable preempt` and `Disconnect dialup interfaces` not supported?

File: plugins/modules/system_high_availability_settings.py [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+    """
+    Handler function for the settings synchronize_config_to_ip,
+    remote_system_username and remote_system_password.
+    Args:
+        config (OPNsenseModuleConfig): The configuration for the opnsense firewall
+        remote_backup_url (Optional[str]): Synchronize your settings to this URL
+        username (Optional[str]): Username for logging in to the remote firewall
+        password (Optional[str]): Password for logging in to the remote firewall
+    """
+    if remote_backup_url:
```
Comment (KiLLuuuhh) [inline plugins/modules/system_high_availability_settings.py:245]: For `remote_backup_url`, `username` and `password` I would suggest that, if the provided value is `""`,  the existing setting is updated respectively cleared.

File: plugins/modules/system_high_availability_settings.py [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+    # check for stderr
+    if result.get("stderr"):
+        raise OPNSenseGetInterfacesError("error encountered while getting services")
+    allowed_services = dict(
+        service.split(",") for service in result.get("stdout_lines")
+    )
+    return allowed_services
+
+
+def services_to_synchronize(config: OPNsenseModuleConfig, sync_services: List[str]):
```
Comment (KiLLuuuhh) [inline plugins/modules/system_high_availability_settings.py:280]: Please add Typing `-> None`

**Review by KiLLuuuhh — CHANGES_REQUESTED** [review]

File: plugins/modules/system_high_availability_settings.py (hunk 1 above)
Comment (LuminatiHD) [reply plugins/modules/system_high_availability_settings.py:351]: I orientated myself closely to the minimum viable product, which didn't include these settings. But there's nothing preventing me from adding them.

Comment (LuminatiHD) [reply plugins/modules/system_high_availability_settings.py:387]: 7ecea2fd1500e9f558b4c566d336676a21487dff

File: plugins/modules/system_high_availability_settings.py (hunk 3 above)
Comment (LuminatiHD) [reply plugins/modules/system_high_availability_settings.py]: a6f4cbcc14c79a63ab51ee904081c5402bc54e29
forgot typing for `synchronize_states` but fixed in 7ecea2fd1500e9f558b4c566d336676a21487dff

File: plugins/modules/system_high_availability_settings.py [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+    """
+    Handler function for the settings synchronize_config_to_ip,
+    remote_system_username and remote_system_password.
+    Args:
+        config (OPNsenseModuleConfig): The configuration for the opnsense firewall
+        remote_backup_url (Optional[str]): Synchronize your settings to this URL
+        username (Optional[str]): Username for logging in to the remote firewall
+        password (Optional[str]): Password for logging in to the remote firewall
+    """
+    if any((remote_backup_url, username, password)):
```
Comment (cfasnacht) [inline plugins/modules/system_high_availability_settings.py:295]: Shouldn't all parameters that are set be checked if they are not null?
Why double brackets?

File: plugins/modules/system_high_availability_settings.py [hunk 5]
```diff
@@ -0,0 +1,489 @@
...
+    # check for stderr
+    if result.get("stderr"):
+        raise OPNSenseGetInterfacesError("error encountered while getting services")
+    allowed_services = dict(
+        service.split(",") for service in result.get("stdout_lines")
+    )
+    return allowed_services
+
+
+def services_to_synchronize(
```
Comment (cfasnacht) [inline plugins/modules/system_high_availability_settings.py:328]: Some comments on what you are doing here would be nice.

File: plugins/modules/system_high_availability_settings.py (hunk 4 above)
Comment (LuminatiHD) [reply plugins/modules/system_high_availability_settings.py:295]: The double brackets are present b/c the function `any()` takes only one parameter, which has to be an iterable. Putting all parameters as is into the function wouldn't work.

File: plugins/modules/system_high_availability_settings.py (hunk 5 above)
Comment (LuminatiHD) [reply plugins/modules/system_high_availability_settings.py:328]: ee795d54e1f0faa36462301ca61f01796edc630a

does that help?

**Review by cfasnacht — APPROVED** [review]

**Review by KiLLuuuhh — APPROVED** [review]

**Pull request merged**